### PR TITLE
[WIP] Store attributes of user classes in separate dictionary

### PR DESCRIPTION
**Pull request by markusschmaus**
## Code review checklist

- [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR)
- [x] Title summarizes what is changing
- [x] Commit messages are meaningful (see [this][commit messages] for details)
- [x] Tests have been included and/or updated
- [ ] Docstrings have been included and/or updated, as appropriate
- [ ] Standalone docs have been updated accordingly
- [ ] Changelog(s) has/have been updated, as needed (see `CHANGELOG.md`, no need
      to update for typo fixes and such).

[commit messages]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: tests/functional/test_user_class_with_slots.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            self.y = y
+
+    modelstr = """
+    shape
+    point 34 45
+    point 12 23
+    end
+    """
+
+    mm = metamodel_from_str(grammar, classes=[Point], auto_init_attributes=False)
```
Comment (igordejanovic) [inline tests/functional/test_user_class_with_slots.py]: Tiny fix is needed here for Travis tests to pass. We are using a max line length of 79 (PEP-8).

**Review by igordejanovic — COMMENTED** [review]
@markusschmaus Great job. This all looks good to me. Just a tiny flake8 fix (see below). I did a performance test comparison and see no regression.

We should probably add a note on this in the part of the docs on user classes, but it is not necessary.

@goto40 Could you please take a look too?

**Comment by goto40** [comment]
I will check this weekend.

**Comment by markusschmaus** [comment]
Supperceded by https://github.com/textX/textX/pull/249

**Pull request closed without merging**